File: windows/AirBridge.Mirror/Interfaces/IMirrorService.cs
using AirBridge.Core.Interfaces;
using AirBridge.Core.Models;

namespace AirBridge.Mirror.Interfaces;

/// <summary>
/// High-level mirror service. Creates and manages mirror sessions.
/// Implemented in Iteration 5/6.
/// </summary>
public interface IMirrorService
{
    /// <summary>
    /// Starts a mirror session with a paired Android device.
    /// <paramref name="mode"/> determines phone-window vs tablet-display.
    /// </summary>
    Task<IMirrorSession> StartMirrorAsync(DeviceInfo remoteDevice, MirrorMode mode, CancellationToken cancellationToken = default);

    /// <summary>Returns all currently active mirror sessions.</summary>
    IReadOnlyList<IMirrorSession> GetActiveSessions();
}

File: android/app/src/main/java/com/airbridge/app/mirror/interfaces/IMirrorService.kt
package com.airbridge.app.mirror.interfaces

import com.airbridge.app.core.interfaces.IMirrorSession
import com.airbridge.app.core.interfaces.MirrorMode
import com.airbridge.app.core.models.DeviceInfo

/**
 * High-level mirror service — manages screen capture sessions.
 * Implemented in Iteration 5/6.
 */
interface IMirrorService {
    suspend fun startMirror(remoteDevice: DeviceInfo, mode: MirrorMode): IMirrorSession
    fun getActiveSessions(): List<IMirrorSession>
}
